pkg/config/validation/validation.go

```diff
@@ -372,15 +372,25 @@ func ValidateHTTPHeaderOperationName(name string) error {
 	return nil
 }
 
+// Copy from https://github.com/bufbuild/protoc-gen-validate/blob/a65858624dd654f2fb306d6af60f737132986f44/module/checker.go#L18
+var httpHeaderValueRegexp = regexp.MustCompile("^[^\u0000-\u0008\u000A-\u001F\u007F]*$")
+
 // ValidateHTTPHeaderValue validates a header value for Envoy
 // Valid: "foo", "%HOSTNAME%", "100%%", "prefix %HOSTNAME% suffix"
-// Invalid: "abc%123"
+// Invalid: "abc%123", "%START_TIME%%"
 // We don't try to check that what is inside the %% is one of Envoy recognized values, we just prevent invalid config.
 // See: https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/headers.html#custom-request-response-headers
 func ValidateHTTPHeaderValue(value string) error {
-	if strings.Count(value, "%")%2 != 0 {
-		return errors.New("single % not allowed.  Escape by doubling to %% or encase Envoy variable name in pair of %")
+	if !httpHeaderValueRegexp.MatchString(value) {
```

Comment:
Not sure I understand this. Its checking if its valid based on HTTP spec, but does it need to be? Envoy substitutes it, so it doesn't need to comply to HTTP it needs to comply to Envoy

Reply:
envoy has proto validator [here](https://github.com/envoyproxy/envoy/blob/8674f661b216ee81ab0b2c868e09830bbe9c7df9/api/envoy/api/v2/core/base.proto#L268)

Reply:
CC @costinm 
We never claimed to support Envoy format strings, did we? I know users rely on it, but they are not subject to strict backwards compatibility guarantees of xDS. I think we may choose to only allow actual values, but that would be a breaking change.

Reply:
For an example of upgrade issue: https://github.com/envoyproxy/envoy/issues/23972

Reply:
IMO, istio's webhook should not block a valid configuration like  `'%DOWNSTREAM_PEER_CERT_V_END(%b %d %H:%M:%S %Y %Z)%'`

Reply:
kindly ping @howardjohn